src/main/webapp/app/entities/submission.model.ts

```diff
@@ -40,3 +40,58 @@ export abstract class Submission implements BaseEntity {
         this.submitted = false; // default value
     }
 }
+
+/**
+ * Used to access the correct submissions result
+ *
+ * @param submission
+ */
+export function getLatestSubmissionResult(submission: Submission | undefined): Result | undefined {
+    checkForResultsLength(submission);
+    if (submission?.results) {
+        const length = submission.results.length;
+        if (length > 0) {
+            return submission.results[length - 1];
+        }
+    }
+}
+
+/**
+ * Used to set the correct submissions result
+ *
+ * @param submission
+ * @param result
+ *
+ * @return the index of the latestSubmissionResult or undefined if any of the params was undefined
+ */
+export function setLatestSubmissionResult(submission: Submission | undefined, result: Result | undefined): number | undefined {
+    checkForResultsLength(submission);
+
+    if (!submission || !result) {
+        return undefined;
+    }
+
+    if (submission.results) {
+        submission.results[submission.results.length - 1] = result;
```

Comment:
What happens with an empty array? Then you would set `submission.results[-1] = result`